pkg/volume/fc/fc_util.go

```diff
@@ -178,8 +187,18 @@ func (util *FCUtil) AttachDisk(b fcDiskMounter) (string, error) {
 }
 
 func (util *FCUtil) DetachDisk(c fcDiskUnmounter, mntPath string) error {
+	devName, _, err := mount.GetDeviceNameFromMount(c.mounter, mntPath)
+	if err != nil {
+		glog.Errorf("fc detach disk: failed to get device from mnt: %s\nError: %v", mntPath, err)
+		return err
+	}
+
 	if err := c.mounter.Unmount(mntPath); err != nil {
 		return fmt.Errorf("fc detach disk: failed to unmount: %s\nError: %v", mntPath, err)
 	}
+	// Remove scsi device from the node.
+	arr := strings.Split(devName, "/")
+	dev := arr[len(arr)-1]
```

Comment:
@rootfs 
I confirmed that the len(arr) is always larger than 1, even if the arr is empty character.
https://play.golang.org/p/08eG3QB7je

Thus I think we don't need worry "panic: runtime error: index out of range' here.